src/Xamarin.Android.Build.Tasks/Utilities/PackagingUtils.cs

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2013 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+ * Original source: https://android.googlesource.com/platform/tools/base/+/d41d662dbf89f9b60ca6256415a059c0107749b8/sdk-common/src/main/java/com/android/ide/common/packaging/PackagingUtils.java
+ */
+
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Xamarin.Android.Tasks
+{
+	/// <summary>
+	/// Utility class for packaging.
+	/// </summary>
+	internal class PackagingUtils
+	{
+		/// <summary>
+		/// The JAR manifest file name.
+		/// http://hg.openjdk.java.net/jdk8u/jdk8u-dev/jdk/file/0fc878b99541/src/share/classes/java/util/jar/JarFile.java#l91
+		/// </summary>
+		const string MANIFEST_NAME = "META-INF/MANIFEST.MF";
```

Comment:
This should be `ManifestFileName` or something more .NET-"normal", as there doesn't appear to be a `MANIFEST_NAME` field in the Java code to be consistent with.

Reply:
Or you could go one further, and do:

```csharp
static readonly string[] InvalidEntryPaths = new[]{
    "META-INF/MANIFEST.MF",
};
```

Then in `CheckEntryForPackaging()`:

```csharp
if (InvalidEntryPaths.Contains (entryName))
    return false;
```